.github/workflows/bikes.yml

```diff
@@ -29,32 +28,70 @@ jobs:
+        echo ::set-output name=result::$branch         
+    
     - name: Create child namespaces and add Dev Spaces labels
       run:  |
-        if [ -n "$(kubectl get namespace $GITHUB_HEAD_REF --ignore-not-found -o name)" ]; then exit 0; fi
-        kubectl create namespace $GITHUB_HEAD_REF
-        kubectl label namespace $GITHUB_HEAD_REF azds.io/parent-space=${{ secrets.MASTER_SPACE }} 
-        kubectl label namespace $GITHUB_HEAD_REF azds.io/space=true
-        kubectl label namespace $GITHUB_HEAD_REF azds.io/pull-request-space=true
+        if [ -n "$(kubectl get namespace ${{steps.generate-child-space-name.outputs.result}} --ignore-not-found -o name)" ]; then exit 0; fi
+        kubectl create namespace ${{steps.generate-child-space-name.outputs.result}}
+        kubectl label namespace ${{steps.generate-child-space-name.outputs.result}} azds.io/parent-space=${{ secrets.MASTER_SPACE }} 
+        kubectl label namespace ${{steps.generate-child-space-name.outputs.result}} azds.io/space=true
+        kubectl label namespace ${{steps.generate-child-space-name.outputs.result}} azds.io/pull-request-space=true
         
     - uses: azure/k8s-actions/k8s-create-secret@master
       with:
-        namespace: ${{ github.head_ref }}
+        namespace: ${{ steps.generate-child-space-name.outputs.result }}
         container-registry-url: ${{ secrets.CONTAINER_REGISTRY }}
         container-registry-username: ${{ secrets.REGISTRY_USERNAME }}
         container-registry-password: ${{ secrets.REGISTRY_PASSWORD }}
         secret-name: ${{ secrets.IMAGE_PULL_SECRET }}
-        
-    - name: Helm Install PR
+
+    - name: Create valid release-name
+      id: generate-release-name
       run: |
-        helm init --upgrade --wait
-        helm upgrade --install --namespace $GITHUB_HEAD_REF bikesharing-$GITHUB_HEAD_REF samples/BikeSharingApp/Bikes/charts/bikes \
+        release=${bikesharing-$GITHUB_HEAD_REF}
+        release=${release::53}
+        release=$(echo ${release//[!0-9a-zA-Z]/-} | tr '[:upper:]' '[:lower:]' | sed -e 's/^-/z/' -e 's/-$/z/')
+        echo ::set-output name=result::$release
+
+    - name: Install Helm for client
+      id: install-helm-client
+      run: |
+        clientversion=$(echo $(helm version --client --short) | sed -n 's/.*Client: \(.*\)+.*/\1/p')
+        serverversion=$(echo $(helm version --server --short) | sed -n 's/.*Server: \(.*\)+.*/\1/p')
+        if [[ ! -n "$serverversion" ]];
+        then
+          echo "Please install Helm on the server!" 1>&2
+          exit 64
+        elif [[ "$clientversion" == "$serverversion" ]]; 
+        then   
+          echo "Versions are equal"
+          echo ::set-output name=result::
+        else
+          mkdir prflow        
+          export DESIRED_VERSION=$serverversion export HELM_INSTALL_DIR=./prflow
+          curl https://raw.githubusercontent.com/helm/helm/master/scripts/get > get_helm.sh
```

Comment:
Write this script to the same temporary directory.

Reply:
I assume that the script execution environment isn't set up to fail fast on commands that return a non-zero exit code. This would mean that even if the curl fails, it would continue to try and run subsequent commands. You should add `set -e` at the start of the script for this step to prevent this.

Reply:
Updated. Non-zero exit code does fail the script though.